app/app.ts

```diff
@@ -396,32 +405,77 @@ function setupIpc(localSettings: LocalSettings, scrSettings: ScrSettings) {
   )
 
   ipcMain.handle('fsReadFile', async (_, filePath) => {
-    return fsPromises.readFile(filePath)
+    try {
+      const arrayBuffer = await fsPromises.readFile(filePath)
+      const result: FsFile = {
+        file: arrayBuffer,
+        error: false,
+      }
+      return result
+    } catch (e) {
+      if ((e as any).code === 'ENOENT') {
+        const error: FsError = {
+          error: true,
+          code: FsErrorCode.ENOENT,
+        }
+        return error
+      }
+      throw e
+    }
   })
   ipcMain.handle('fsReadDir', async (_, dirPath, options) => {
-    const result = await fsPromises.readdir(dirPath, options)
-    return result.map<FsDirent>(d => ({
-      isFile: d.isFile(),
-      isDirectory: d.isDirectory(),
-      name: d.name,
-    }))
+    try {
+      const result = await fsPromises.readdir(dirPath, options)
+      const files = result.map<FsDirent>(d => ({
+        isFile: d.isFile(),
+        isDirectory: d.isDirectory(),
+        name: d.name,
+      }))
+      const dirents: FsDirents = {
+        files,
+        error: false,
```

Comment:
If you make the `error` property optional on the result type, then you can omit this.